client/iotcore/model.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package iotcore
+
+import (
+	"bytes"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+
+	"github.com/mendersoftware/iot-manager/model"
+)
+
+type Device struct {
+	ID            string `json:"id"`
+	Name          string `json:"name"`
+	Version       int64  `json:"version,omitempty"`
+	Status        Status `json:"status,omitempty"`
+	CertificateID string `json:"certificate_id,omitempty"`
+	Certificate   string `json:"certificate,omitempty"`
+	PrivateKey    string `json:"private_key,omitempty"`
+}
```

Comment:
shouldn't we validate `Device` structure?
security sweep: how long does PrivateKey stay in memory in plain format?

Reply:
No need to validate this struct, as we never marshall/unmarshall it directly.
This is a holder of data we get/set from/to the SDK.